### PR TITLE
Add lax implementation of np.isin() and np.in1d()

### PR DESCRIPTION
**Pull request by jakevdp**
I did not use a `searchsorted()` based implementation, because it does not beat the simple approach on CPU until both arrays are larger than ~10,000 elements.

A hash-based solution may be better for large inputs, but I'm not sure whether XLA has fast hash tables.

Alternatively, we could co-sort unique values from each array and check for adjacent duplicates in the result; this approach would have to wait for ``jnp.unique()`` and ``jnp.argsort()`` to support masking.

#2078

### CODE REVIEW

**Comment by mattjj** [comment]
Nice!

XLA does not have hash tables. It's just arrays! (Well, and tuples, but we almost never use those. And tokens, but those are pretty special-purpose.)

**Review by mattjj — APPROVED** [review]
LGTM!

**Comment by jakevdp** [comment]
Internal tests pass; merging.
